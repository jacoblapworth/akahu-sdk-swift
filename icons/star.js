export default "M15 6l2.781 5.926 6.219.949-4.5 4.613L20.562 24 15 20.925 9.438 24l1.062-6.512L6 12.875l6.219-.949 Z";