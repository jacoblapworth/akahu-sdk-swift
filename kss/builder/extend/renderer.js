const { regexDictionary, types } = require('../parser/constants');
const tinycolor = require('tinycolor2');

module.exports = function(handlebars) {
	/**
	 * Extracts the variable properties from the line
	 * returns:
	 *  token name, token value, and optional [ token example ]
	 */
	handlebars.registerHelper('renderTokens', renderTokens);

	/**
	 * Compares the name to see if the container needs to be rendered as inverted.
	 * Similar to the isEqual
	 *
	 * e.g.
	 * {{#isInverted name}}
	 * 		ADD THIS
	 * {{else}} [optional]
	 * 		WILL EXECUTE THIS
	 * {{/isInverted}}
	 */
	handlebars.registerHelper('isInverted', isInverted);
};

const testString = 'To understand what recursion is, you must first understand recursion.';


function renderTokens(variables, block) {
	var newBlock = '';
	variables.split('\n').forEach(line => {
		var test;
		const matches = line.match(/[^:]+/g).map(match => match.trim());

		if (matches[1]) {
			this.token = {};
			this.token.name = matches[0];
			this.token.value = matches[1];
			this.token.title = matches[3];
			this.token.description = matches[4];
			if (matches[2] != undefined) {
				this.token.example = getExample(matches[2], this.token);
			}
			newBlock += block.fn(this);
		}
	});
	return newBlock;
}

function getExample(type, token) {
	switch(type) {
		case types.color:
			const codeTextClass = tinycolor(token.value).isDark() ?
				'ds-colour-code-light'
				: null;
			return `
					<div class="ds-colour-example" style="background-color:${token.value}">
						<code class="${codeTextClass}">${token.value}</code>
						<code class="${codeTextClass}">${token.name}</code>
					</div>`;
		case types.border:
			return `
				<div class="ds-colour-example" style="border:${token.value}">
					<code>${token.name}</code>
				</div>`;
		case types.shadow:
			return `
					<div class="ds-colour-example" style="box-shadow:${token.value}">
						<code>${token.name}</code>
					</div>`;
		case types.fontSize:
			return  `<span class="xui-text-truncated" style="font-size:${token.value};">Aa</span>`;
		case types.fontWeight:
			return `<span style="font-weight:${token.value};">Aa</span>`;
		case types.lineHeight:
			return `<span style="line-height:${token.value}">${testString}</span>`;
		case types.spacing:
			return `<div class="ds-spacing-example" style="height:${token.value};width:${token.value};"></div>`;
		default:
			return '';
	}
}

function isInverted(name, options) {
	return regexDictionary.inverted.test(name) ? options.fn(this) : options.inverse(this);
}
