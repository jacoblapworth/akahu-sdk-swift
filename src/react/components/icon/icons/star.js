export default 'M 15.5,18.5 10.5,21 11.333333,15.583333 8,11.833333 13,11 l 2.5,-5 2.5,5 5,0.833333 -3.333333,3.75 L 20.5,21 Z';