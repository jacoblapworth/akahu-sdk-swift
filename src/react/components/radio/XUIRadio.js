import React, { createRef, useEffect } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import '../helpers/xuiGlobalChecks';
import { baseClass } from './constants';
import { ns } from '../helpers/xuiClassNamespace';
import XUIControlWrapperInline from '../controlwrapper/XUIControlWrapperInline';
import generateIds, { getAriaAttributesInline } from '../helpers/ariaHelpers';
import XUITouchTarget from '../touchtarget/XUITouchTarget';
import labelRequiredError, { textChildOrLabelId } from '../helpers/labelRequiredError';
import shouldRender from '../helpers/shouldRender';

/**
 * @function handleLabelClick - Prevent 2 click events bubbling. Since our input is
 * wrapped inside a label, then clicking the label will also cause a new click event
 * on the input, which also bubbles up. If a consumer attaches an onClick event listener
 * further up the DOM, we don't want it to be fired twice.
 *
 * @private
 */
const onLabelClick = e => {
  if (e.target.tagName !== 'INPUT') {
    e.stopPropagation();
  }
};

/**
 * @function buildSvgRadio - If triggered with a custom icon path, build svg radio
 * @param qaHook - Optional hook label
 * @param svgSettings - Object containing optional svg properties (classname, icon paths)
 *
 */
const buildSvgRadio = (qaHook, { radioElementClassName, iconMain }) => {
  const svgClasses = cn(`${ns}-icon`, radioElementClassName);
  return (
    <div className={`${ns}-iconwrapper`}>
      <svg
        className={svgClasses}
        data-automationid={qaHook && `${qaHook}--icon`}
        height={iconMain.height}
        viewBox={`0 0 ${iconMain.width} ${iconMain.height}`}
        width={iconMain.width}
      >
        <path className={`${baseClass}--focus-outer`} d={iconMain.path} role="presentation" />
        <path className={`${baseClass}--focus`} d={iconMain.path} role="presentation" />
        <path className={`${baseClass}--main`} d={iconMain.path} role="presentation" />
      </svg>
      <XUITouchTarget />
    </div>
  );
};

/**
 * @function buildHtmlRadio - build the HTML version of the radio control
 * @param qaHook - Optional hook label
 * @param className - Optional classname to add to html version of radio
 * @param calculatedSize - String to specify the size of the radio
 *
 */
const buildHtmlRadio = (qaHook, className, calculatedSize) => {
  const htmlClasses = cn(
    `${baseClass}--radio`,
    className,
    calculatedSize && `${baseClass}--radio-${calculatedSize}`,
  );
  return (
    <div className={htmlClasses} data-automationid={qaHook && `${qaHook}--radio`}>
      <XUITouchTarget />
    </div>
  );
};

/**
 * @function buildRadio - given the radio props supplied, select which radio
 * builder to trigger
 * @param qaHook - Optional hook label
 * @param htmlClassName - Optional classname to add to html version of radio
 * @param svgSettings - Object containing optional svg properties (classname, icon paths)
 * @param calculatedSize - String to specify the size of the radio
 *
 */
const buildRadio = (qaHook, htmlClassName, svgSettings, calculatedSize) => {
  if (svgSettings.iconMain) {
    return buildSvgRadio(qaHook, svgSettings);
  }
  return buildHtmlRadio(qaHook, htmlClassName, calculatedSize);
};

const XUIRadio = props => {
  // User can manually provide an id, or we will generate one.
  const wrapperIds = generateIds({ labelId: props.labelId, id: props.id });
  const {
    children,
    className,
    hintMessage,
    iconMain,
    id,
    inputProps: radioInputProps,
    isChecked,
    isDefaultChecked,
    isDisabled,
    isGrouped,
    isInvalid,
    isLabelHidden,
    isRequired,
    isReversed,
    labelClassName,
    name,
    onChange,
    qaHook,
    radioElementClassName,
    role,
    size,
    tabIndex,
    validationMessage,
    value,
  } = props;

  const labelRef = createRef();

  // Grouped inputs default to 'small'.
  const calculatedSize = (isGrouped && 'small') || size;

  const classes = cn(
    baseClass,
    isReversed && shouldRender(children) && !isLabelHidden && `${baseClass}-reversed`,
    isDisabled && `${ns}-styledcheckboxradio-is-disabled`,
  );

  const wrapperClasses = cn(
    className,
    `${baseClass}wrapper`,
    calculatedSize && `${baseClass}-${calculatedSize}`,
  );

  const labelClasses = cn(
    `${baseClass}--label`,
    calculatedSize && `${baseClass}--label-${calculatedSize}`,
    labelClassName,
  );

  const inputProps = {
    ...radioInputProps,
    type: 'radio',
    disabled: isDisabled,
    required: isRequired,
    tabIndex,
    name,
    onChange,
    value,
    id,
    ...getAriaAttributesInline(wrapperIds, props),
  };
  const svgSettings = {
    iconMain,
    radioElementClassName,
  };

  useEffect(() => {
    labelRequiredError(XUIRadio.name, textChildOrLabelId, [
      labelRef.current?.textContent && !isLabelHidden,
      typeof children?.[0] === 'string',
      props.labelId,
    ]);
  }, [children, isLabelHidden, labelRef, props.labelId]);

  if (typeof isChecked !== 'boolean') {
    inputProps.defaultChecked = !!isDefaultChecked;
  } else {
    inputProps.checked = isChecked;
    // checked prop without an onChange handler means this is readonly, so set that to prevent
    // warnings in the console.
    if (onChange == null) {
      inputProps.readOnly = true;
    }
  }

  return (
    <XUIControlWrapperInline
      fieldClassName={classes}
      label={children}
      labelClassName={labelClasses}
      labelRef={labelRef}
      messageClassName={`${baseClass}--message`}
      onClick={onLabelClick}
      rootClassName={wrapperClasses}
      wrapperIds={wrapperIds}
      {...{
        qaHook,
        isInvalid,
        validationMessage,
        hintMessage,
        isLabelHidden,
      }}
    >
      <input
        className={cn(
          `${baseClass}--input`,
          inputProps.className,
          calculatedSize && `${baseClass}--input-${calculatedSize}`,
          radioElementClassName,
        )}
        data-automationid={qaHook && `${qaHook}--input`}
        role={role}
        {...inputProps}
      />
      {buildRadio(qaHook, radioElementClassName, svgSettings, calculatedSize)}
    </XUIControlWrapperInline>
  );
};

export default XUIRadio;

XUIRadio.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,

  /** Hint message to show under the input */
  hintMessage: PropTypes.node,

  /** The icon path to use for the radio */
  iconMain: PropTypes.shape({
    height: PropTypes.number.isRequired,
    path: PropTypes.string.isRequired,
    width: PropTypes.number.isRequired,
  }),

  id: PropTypes.string,
  /** Props to be spread onto the radio type of the input element itself */
  inputProps: PropTypes.object,

  /** The input is selected */
  isChecked: PropTypes.bool,

  /** Used to output an uncontrolled radio component.  If a value is passed to the
   * isChecked prop, this prop will be ignored. */
  isDefaultChecked: PropTypes.bool,

  /** The input is disabled */
  isDisabled: PropTypes.bool,

  /** Used by XUI components to state whether the radio is part of a group */
  isGrouped: PropTypes.bool,

  /** Whether the current input value is invalid */
  isInvalid: PropTypes.bool,

  /** Prevents the label element from being displayed on the page. Label is still
   * accessible to screen readers. */
  isLabelHidden: PropTypes.bool,

  /** The input is required for form submission */
  isRequired: PropTypes.bool,

  /** The label and control are displayed in reverse order */
  isReversed: PropTypes.bool,

  /** Additional class names on the span (pseudo-label) element  */
  labelClassName: PropTypes.string,

  /** Provide a specific label ID which will be used as the "labelleby" aria property */
  labelId: PropTypes.string,

  /** The name to use as a reference for the value */
  name: PropTypes.string,

  /** onChange - The function to call when the control changes state */
  onChange: PropTypes.func,

  qaHook: PropTypes.string,

  /** Additional class names for the input and html/svg elements */
  radioElementClassName: PropTypes.string,

  /** Role to be applied for screen readers */
  role: PropTypes.string,

  /** Size variant. Defaults to medium */
  size: PropTypes.oneOf(['medium', 'small', 'xsmall']),

  /** The tabindex property to place on the radio input */
  tabIndex: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),

  /** Validation message to show under the input if `isInvalid` is true */
  validationMessage: PropTypes.node,

  /** The value to return on form submission */
  value: PropTypes.string,
};

XUIRadio.defaultProps = {
  isDisabled: false,
  isLabelHidden: false,
  isRequired: false,
  isReversed: false,
  role: 'radio',
  size: 'medium',
};
