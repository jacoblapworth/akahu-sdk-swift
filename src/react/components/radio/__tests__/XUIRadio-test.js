import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import XUIRadio from '../XUIRadio';
import radioMain from '@xero/xui-icon/icons/radio-main';
import radioCheck from '@xero/xui-icon/icons/radio-check';
import checkboxCheck from '@xero/xui-icon/icons/checkbox-check';
import star from '@xero/xui-icon/icons/star';

Enzyme.configure({ adapter: new Adapter() });

const NOOP = () => {};

describe('XUIRadio', () => {

	// type radio
	it('should be of type radio', () => {
		const component = shallow(<XUIRadio onChange={NOOP}/>);

		expect(component.find('input').prop('type')).toEqual('radio');
	});

	// styled div for html radio
	it('should contain a styled div, by default', () => {
		const component = shallow(<XUIRadio onChange={NOOP}/>);

		expect(component.find('div').hasClass('xui-styledcheckboxradio--radio')).toBeTruthy();
	});


	// children property (label text)
	it('should have label text if provided', () => {
		const component = shallow(<XUIRadio onChange={NOOP}>Howdy, folks!</XUIRadio>);

		expect(component.text()).toEqual('Howdy, folks!');
	});


	// className property (additional classes)
	it('should use additional classes on the root node if provided', () => {
		const component = shallow(<XUIRadio onChange={NOOP} className="dogs-are-totes-patotes" />);

		expect(component.hasClass('dogs-are-totes-patotes')).toBeTruthy();
	});


	// qaHook property
	it.skip('should have a qaHook on the root node if provided', () => {
		const component = renderer.create(<XUIRadio onChange={NOOP} qaHook="cheese-and-crackers" />);

		expect(component).toMatchSnapshot();
	});

	it('should have a hidden label, if specified', () => {
		const hiddenLabel = renderer.create(
			<XUIRadio onChange={NOOP} isLabelHidden>Hidden label</XUIRadio>
		);

		expect(hiddenLabel).toMatchSnapshot();
	});

	describe('icon combinations', () => {
		const defaultComponent = shallow(<XUIRadio onChange={NOOP} iconMainPath={radioMain} iconCheckPath={radioCheck}/>);
		// SVG element uses xui-icon class
		it('should use the xui-icon class on the SVG element', () => {
			expect(defaultComponent.find('svg').hasClass('xui-icon')).toBeTruthy();
		});


		// SVG path elements include role presentation
		it('should define role as presentation on each path element', () => {
			expect(defaultComponent.find('svg').childAt(0).prop('role')).toEqual('presentation')
			expect(defaultComponent.find('svg').childAt(1).prop('role')).toEqual('presentation')
			expect(defaultComponent.find('svg').childAt(2).prop('role')).toEqual('presentation')

		});


		// Icon combinatorics 1/9 (main: undefined, check: undefined)
		it('should not generate svgs if iconMainPath and iconCheckPath if both are undefined', () => {
			const component = shallow(<XUIRadio onChange={NOOP} />);

			expect(component.find('svg').length).toEqual(0);
		});


		// Icon combinatorics 2/9 (main: undefined, check: null)
		it('should not generate svgs if iconMainPath is undefined and iconCheckPath is null', () => {
			const component = shallow(<XUIRadio onChange={NOOP} iconCheckPath={null} />);

			expect(component.find('svg').length).toEqual(0);
		});


		// Icon combinatorics 3/9 (main: undefined, check: checkbox-check)
		it('should use the default iconMainPath and a custom iconCheckPath if iconMainPath is undefined and iconCheckPath is defined', () => {
			const component = shallow(<XUIRadio onChange={NOOP} iconCheckPath={checkboxCheck} />);

			expect(component.find('svg').childAt(0).prop('d')).toEqual(radioMain);
			expect(component.find('svg').childAt(1).prop('d')).toEqual(radioMain);
			expect(component.find('svg').childAt(2).prop('d')).toEqual(checkboxCheck);
		});


		// Icon combinatorics 4/9 (main: null, check: undefined)
		it('should not generate svgs if iconMainPath is null and iconCheckPath is undefined', () => {
			const component = shallow(<XUIRadio onChange={NOOP} iconMainPath={null} />);

			expect(component.find('svg').length).toEqual(0);
		});


		// Icon combinatorics 5/9 (main: null, check: null)
		it('should not generate svgs if both are null', () => {
			const component = shallow(<XUIRadio onChange={NOOP} iconMainPath={null} iconCheckPath={null} />);

			expect(component.find('svg').length).toEqual(0);
		});


		// Icon combinatorics 6/9 (main: null, check: checkbox-check)
		it('should use the default iconMainPath and a custom iconCheckPath if iconMainPath is null and iconCheckPath is defined', () => {
			const component = shallow(<XUIRadio onChange={NOOP} iconMainPath={null} iconCheckPath={checkboxCheck} />);

			expect(component.find('svg').childAt(0).prop('d')).toEqual(radioMain);
			expect(component.find('svg').childAt(1).prop('d')).toEqual(radioMain);
			expect(component.find('svg').childAt(2).prop('d')).toEqual(checkboxCheck);
		});


		// Icon combinatorics 7/9 (main: star, check: undefined)
		it('should use a custom iconMainPath without a checkmark if iconMainPath is defined and iconCheckPath is undefined', () => {
			const component = shallow(<XUIRadio onChange={NOOP} iconMainPath={star} />);

			expect(component.find('svg').childAt(0).prop('d')).toEqual(star);
			expect(component.find('svg').childAt(1).prop('d')).toEqual(star);
			expect(component.find('svg').children().length).toEqual(2); //No checkmark path element
		});


		// Icon combinatorics 8/9 (main: star, check: null)
		it('should use a custom iconMainPath without a checkmark if iconMainPath is defined and iconCheckPath is null', () => {
			const component = shallow(<XUIRadio onChange={NOOP} iconMainPath={star} iconCheckPath={null} />);

			expect(component.find('svg').childAt(0).prop('d')).toEqual(star);
			expect(component.find('svg').childAt(1).prop('d')).toEqual(star);
			expect(component.find('svg').children().length).toEqual(2); //No checkmark path element
		});


		// Icon combinatorics 9/9 (main: star, check: checkbox-check)
		it('should use a custom iconMainPath and a custom iconCheckPath if both are defined', () => {
			const component = shallow(<XUIRadio onChange={NOOP} iconMainPath={star} iconCheckPath={checkboxCheck} />);

			expect(component.find('svg').childAt(0).prop('d')).toEqual(star);
			expect(component.find('svg').childAt(1).prop('d')).toEqual(star);
			expect(component.find('svg').childAt(2).prop('d')).toEqual(checkboxCheck);
		});
	});


	// Unchecked
	it('should be unchecked by default', function () {
		const component = shallow(<XUIRadio onChange={NOOP} />);

		expect(component.html()).not.toContain('checked');
	});


	// isChecked property
	it('should be selected if isChecked is true', function () {
		const component = shallow(<XUIRadio onChange={NOOP} isChecked={true} />);

		expect(component.html()).toContain('checked');
	});


	// isDisabled property
	it('should be disabled if isDisabled is true', function () {
		const component = shallow(<XUIRadio onChange={NOOP} isDisabled={true} />);

		expect(component.html()).toContain('disabled');
	});


	// isChecked and isDisabled properties
	it('should be selected and disabled if isChecked and isDisabled are both true', function () {
		const component = shallow(<XUIRadio onChange={NOOP} isChecked={true} isDisabled={true} />);

		expect(component.html()).toContain('disabled');
		expect(component.html()).toContain('checked');
	});


	// isRequired property
	it('should be required for form submission if isRequired is true', function () {
		const component = shallow(<XUIRadio onChange={NOOP} isRequired={true} />);

		expect(component.html()).toContain('required');
	});


	// isReversed property
	it('should use the xui-styledcheckboxradio-reverse class on the root node if isReversed is true', function () {
		const component = shallow(<XUIRadio onChange={NOOP} isReversed={true} />);

		expect(component.hasClass('xui-styledcheckboxradio-reversed')).toBeTruthy();
	});


	// name property
	it('should have the correct name if one is provided', function () {
		const component = shallow(<XUIRadio onChange={NOOP} name="Patrick" />);

		expect(component.childAt(0).prop('name')).toEqual('Patrick');
	});


	// onChange property
	it('should call the provided onChange function every time the control changes state', function () {
		const callback = jest.fn();
		const component = shallow(<XUIRadio onChange={callback} />);

		component.find('input').simulate('change');
		expect(callback.mock.calls.length).toEqual(1);

	});


	// value property
	it('should have the correct value if one is provided', function () {
		const component = shallow(<XUIRadio onChange={NOOP} value="64" />);

		expect(component.childAt(0).prop('value')).toEqual('64');
	});

});
