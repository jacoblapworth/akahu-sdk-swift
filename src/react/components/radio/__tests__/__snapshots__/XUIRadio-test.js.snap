// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XUIRadio should be a small variant, if specified 1`] = `
<div
  className="xui-styledcheckboxradiowrapper xui-styledcheckboxradio-small"
>
  <label
    className="xui-styledcheckboxradio"
    onClick={[Function]}
  >
    <input
      aria-labelledby="xui-testRadioId"
      className="xui-styledcheckboxradio--input xui-styledcheckboxradio--input-small"
      defaultChecked={false}
      disabled={false}
      onChange={[Function]}
      required={false}
      role="radio"
      type="radio"
    />
    <div
      className="xui-styledcheckboxradio--radio xui-styledcheckboxradio--radio-small"
    >
      <div
        className="xui-touchtarget"
      />
    </div>
    <span
      className="xui-styledcheckboxradio--label xui-styledcheckboxradio--label-small"
      id="xui-testRadioId"
    >
      Howdy, folks!
    </span>
  </label>
</div>
`;

exports[`XUIRadio should be a xsmall variant, if specified 1`] = `
<div
  className="xui-styledcheckboxradiowrapper xui-styledcheckboxradio-xsmall"
>
  <label
    className="xui-styledcheckboxradio"
    onClick={[Function]}
  >
    <input
      aria-labelledby="xui-testRadioId"
      className="xui-styledcheckboxradio--input xui-styledcheckboxradio--input-xsmall"
      defaultChecked={false}
      disabled={false}
      onChange={[Function]}
      required={false}
      role="radio"
      type="radio"
    />
    <div
      className="xui-styledcheckboxradio--radio xui-styledcheckboxradio--radio-xsmall"
    >
      <div
        className="xui-touchtarget"
      />
    </div>
    <span
      className="xui-styledcheckboxradio--label xui-styledcheckboxradio--label-xsmall"
      id="xui-testRadioId"
    >
      Howdy, folks!
    </span>
  </label>
</div>
`;

exports[`XUIRadio should have a hidden label, if specified 1`] = `
<div
  className="xui-styledcheckboxradiowrapper xui-styledcheckboxradio-medium"
>
  <label
    className="xui-styledcheckboxradio"
    onClick={[Function]}
  >
    <input
      aria-label="Hidden label"
      className="xui-styledcheckboxradio--input xui-styledcheckboxradio--input-medium"
      defaultChecked={false}
      disabled={false}
      onChange={[Function]}
      required={false}
      role="radio"
      type="radio"
    />
    <div
      className="xui-styledcheckboxradio--radio xui-styledcheckboxradio--radio-medium"
    >
      <div
        className="xui-touchtarget"
      />
    </div>
  </label>
</div>
`;

exports[`XUIRadio should have a qaHook on the root node if provided 1`] = `
<div
  className="xui-styledcheckboxradiowrapper xui-styledcheckboxradio-medium"
>
  <label
    className="xui-styledcheckboxradio"
    data-automationid="cheese-and-crackers"
    onClick={[Function]}
  >
    <input
      className="xui-styledcheckboxradio--input xui-styledcheckboxradio--input-medium"
      data-automationid="cheese-and-crackers--input"
      defaultChecked={false}
      disabled={false}
      onChange={[Function]}
      required={false}
      role="radio"
      type="radio"
    />
    <div
      className="xui-styledcheckboxradio--radio xui-styledcheckboxradio--radio-medium"
      data-automationid="cheese-and-crackers--radio"
    >
      <div
        className="xui-touchtarget"
      />
    </div>
  </label>
</div>
`;

exports[`XUIRadio should have label text if provided 1`] = `
<div
  className="xui-styledcheckboxradiowrapper xui-styledcheckboxradio-medium"
>
  <label
    className="xui-styledcheckboxradio"
    onClick={[Function]}
  >
    <input
      aria-labelledby="xui-testRadioId"
      className="xui-styledcheckboxradio--input xui-styledcheckboxradio--input-medium"
      defaultChecked={false}
      disabled={false}
      onChange={[Function]}
      required={false}
      role="radio"
      type="radio"
    />
    <div
      className="xui-styledcheckboxradio--radio xui-styledcheckboxradio--radio-medium"
    >
      <div
        className="xui-touchtarget"
      />
    </div>
    <span
      className="xui-styledcheckboxradio--label xui-styledcheckboxradio--label-medium"
      id="xui-testRadioId"
    >
      Howdy, folks!
    </span>
  </label>
</div>
`;

exports[`XUIRadio should render with an icon when one is provided 1`] = `
<div
  className="xui-styledcheckboxradiowrapper xui-styledcheckboxradio-medium"
>
  <label
    className="xui-styledcheckboxradio"
    onClick={[Function]}
  >
    <input
      className="xui-styledcheckboxradio--input xui-styledcheckboxradio--input-medium"
      defaultChecked={false}
      disabled={false}
      onChange={[Function]}
      required={false}
      role="radio"
      type="radio"
    />
    <div
      className="xui-iconwrapper"
    >
      <svg
        className="xui-icon"
        height={15}
        viewBox="0 0 15 15"
        width={15}
      >
        <path
          className="xui-styledcheckboxradio--focus-outer"
          d="M7.5 12.5l-5 2.5.833-5.417L0 5.833 5 5l2.5-5L10 5l5 .833-3.333 3.75L12.5 15z"
          role="presentation"
        />
        <path
          className="xui-styledcheckboxradio--focus"
          d="M7.5 12.5l-5 2.5.833-5.417L0 5.833 5 5l2.5-5L10 5l5 .833-3.333 3.75L12.5 15z"
          role="presentation"
        />
        <path
          className="xui-styledcheckboxradio--main"
          d="M7.5 12.5l-5 2.5.833-5.417L0 5.833 5 5l2.5-5L10 5l5 .833-3.333 3.75L12.5 15z"
          role="presentation"
        />
      </svg>
      <div
        className="xui-touchtarget"
      />
    </div>
  </label>
</div>
`;

exports[`XUIRadio should render with validation message 1`] = `
<div
  className="xui-styledcheckboxradiowrapper xui-styledcheckboxradio-medium"
>
  <label
    className="xui-styledcheckboxradio"
    onClick={[Function]}
  >
    <input
      aria-describedby="xui-testRadioId-message"
      aria-invalid={true}
      className="xui-styledcheckboxradio--input xui-styledcheckboxradio--input-medium"
      defaultChecked={false}
      disabled={false}
      required={false}
      role="radio"
      type="radio"
    />
    <div
      className="xui-styledcheckboxradio--radio xui-styledcheckboxradio--radio-medium"
    >
      <div
        className="xui-touchtarget"
      />
    </div>
  </label>
  <div
    className="xui-validation xui-validation-layout xui-validation-is-invalid xui-styledcheckboxradio--message"
    id="xui-testRadioId-message"
    role="status"
  >
    <svg
      className="xui-icon"
      focusable="false"
      height={16}
      viewBox="0 0 16 16"
      width={16}
    >
      <path
        d="M8,16 C3.581722,16 0,12.418278 0,8 C0,3.581722 3.581722,0 8,0 C12.418278,0 16,3.581722 16,8 C16,12.418278 12.418278,16 8,16 Z M7,4 L7,9 L9,9 L9,4 L7,4 Z M7,10 L7,12 L9,12 L9,10 L7,10 Z"
        role="presentation"
      />
    </svg>
    Test validation
  </div>
</div>
`;
